logstash-core/lib/logstash/config/mixin.rb

```diff
@@ -513,6 +514,12 @@ def validate_value(value, validator)
             end
 
             result = value.first.is_a?(::LogStash::Util::Password) ? value.first : ::LogStash::Util::Password.new(value.first)
+          when :uri
+            if value.size > 1
+              return false, "Expected uri (one value), got #{value.size} values?"
+            end
+            
+            result = value.first.is_a?(::LogStash::Util::SafeURI) ? value.first : ::LogStash::Util::SafeURI.new(value.first)
```

Comment:
Weird stuff happens during params parsing and values can be reparsed. I lifted this code from the Password validation which does the same thing. See https://github.com/andrewvc/logstash/blob/5956d8ecd089ef348c9b937ee8a06bcb48eb3221/logstash-core/lib/logstash/config/mixin.rb#L511